### ISSUE TITLE
Tables: ImGuiSelectableFlags_SpanAllColumns do not clip properly with frozen header row

### ISSUE THREAD

**Issue by jpbrault** [ocornut/imgui#3479]
**Version/Branch of Dear ImGui:**

Version: 1.78 WIP
Branch: table

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_SDL.cpp + imgui_impl_OpenGL2.cpp _
Compiler: GCC 9.0 / Clang 10.1
Operating System: Linux / Mac

**My Issue/Question:**

When a selectable is used in a table with the ImGuiSelectableFlags_SpanAllColumns, it will not be clipped properly under the header row if ImGuiTableFlags_ScrollFreezeTopRow is enabled.

**Screenshots/Video**

![problem2](https://user-images.githubusercontent.com/8876173/93618434-445bac00-f9a5-11ea-9a57-8103700d5f3a.gif)

```
if ( ImGui::Begin( "Window" ) )
{
  static bool spanAllColumn{ true };
  ImGui::Checkbox("ImGuiSelectableFlags_SpanAllColumns", & spanAllColumn );
  
  if ( ImGui::BeginTable( "Table", 3,
         ImGuiTableFlags_ScrollFreezeTopRow |
           ImGuiTableFlags_ScrollFreezeLeftColumn |
           ImGuiTableFlags_RowBg |
           ImGuiTableFlags_Scroll |
           ImGuiTableFlags_ScrollY ) )
  {
    ImGui::TableSetupColumn( "Column 1" );
    ImGui::TableSetupColumn( "Column 2" );
    ImGui::TableSetupColumn( "Column 3" );
    ImGui::TableAutoHeaders();

    const int nbLines = 30;
    for ( int i = 0; i < nbLines; ++i )
    {
      ImGui::TableNextRow();
      ImGui::PushID( i );

      ImGui::TableSetColumnIndex( 0 );

      ImGui::Selectable( "value 1", i % 2,
                          spanAllColumn ? ImGuiSelectableFlags_SpanAllColumns
                                        : ImGuiSelectableFlags_None );
      if ( ImGui::TableNextCell() )
      {
        ImGui::Text( "value 2" );
      }
      if ( ImGui::TableNextCell() )
      {
        ImGui::Text( "value 3" );
      }

      ImGui::PopID();
    }
  }
  ImGui::EndTable();
}
ImGui::End();
```

**Comment by ocornut** [comment]
Hello,
That's a known issue, there's an easy/dumb fix but I'm unhappy with it was it would break some of the draw call merging code I have in place, so I'm waiting until I can implement a more satisfying solution. My intuition is I should refactor draw channels allocation/assignment completely, as there are other issues (with borders rendering) also in need of that.

**Comment by ocornut** [comment]
For reference, you can see this in the demo under Tables->Advanced, if you click Options->Other->ContentsType and choose Selectable:
![image](https://user-images.githubusercontent.com/8225057/93619388-c7e2c080-f9d8-11ea-9ed6-466053e55260.png)

In `PushTableBackground()` if you replace
```
SetWindowClipRectBeforeSetChannel(window, table->HostClipRect);
```
with
```
SetWindowClipRectBeforeSetChannel(window, table->BackgroundClipRect);
```
It will fix the visible bug, however if your `Selectable()` are not empty or RowBg are used this will create many extraneous draw calls, so I'm not happy with that solution.

**Comment by jpbrault** [comment]
Thanks for the workaround.

It seem acceptable in my use case.

**Comment by jpbrault** [comment]
The issue seem to be have been fixed in latest commit from October 7th.

The line in PushTableBackground() is now

```
SetWindowClipRectBeforeSetChannel(window, table->HostClipRect);
```

Thanks for the fix.

**Comment by ocornut** [comment]
This should now be fixed by 0568ac25757b640afd989f5bdca9ac17783b5364.

**Comment by fatal338** [comment]
hey can u make a tutorial on how to make this gui? on mac
